### DISCUSSION TITLE
imgui::image corrupted texture... Have any idea?

### DISCUSSION THREAD

**Discussion by mbeytekin** [ocornut/imgui#4048] [Old/Archived discussions]
I don't think my problem is directly related to imgui. But despite dealing with this issue for almost a week, I could not get any results. So I would be glad if you could give me an idea of where I went wrong.

I am trying to use imgui in a dll I have created. I access the functions related to the main program and make the process. But when it comes to displaying images, I can't do anything.

I am using dx11 backend. I created a texture by following the steps in https://github.com/ocornut/imgui/wiki/Image-Loading-and-Displaying-Examples. Then again as in this example;
          `ImGui :: Begin ("DirectX11 Texture Test");
          ImGui :: Text ("pointer =% p", my_texture);
          ImGui :: Text ("size =% d x% d", my_image_width, my_image_height);
          ImGui :: Image ((void *) my_texture, ImVec2 (my_image_width, my_image_height));
          ImGui :: End ();
          `

The image appears, but its content shows a dynamic region, although its dimensions are correct.

The image it shows is the host program's own content .. and as it moves the imgui window, it displays the things in the host program's window by scaling the size of the image I want to display in the imgui :: image section.

The part where I activate imgui from this host program:

                `fn_export double imgui_setup (char * hwnd, char * device, char * device_context) {
                
                // setup context
                IMGUI_CHECKVERSION ();
                ImGui :: CreateContext ();
                
                // Enable keyboard navigation, docking, and viewports
                ImGuiIO & io = ImGui :: GetIO (); (void) io;
                io.ConfigFlags | = ImGuiConfigFlags_NavEnableKeyboard;
                //io.ConfigFlags | = ImGuiConfigFlags_DockingEnable;
                //io.ConfigFlags | = ImGuiConfigFlags_ViewportsEnable;
                
                // setup style
                ImGui :: StyleColorsDark ();
                
                // tweak WindowRounding / WindowBg so platform windows can look identical to regular ones.
                ImGuiStyle & style = ImGui :: GetStyle ();
                style.WindowRounding = 8.0f;
                style.Colors [ImGuiCol_WindowBg] .w = 1.0f;
                
                CreateDeviceD3D ((HWND) hwnd);
                
                // setup platform / renderer bindings
                ImGui_ImplWin32_Init ((void *) hwnd);
                ImGui_ImplDX11_Init ((ID3D11Device *) device, (ID3D11DeviceContext *) device_context);
                
                // Subclass GM window to update keyboard / mouse / etc events
                SetWindowSubclass ((HWND) hwnd, ImGuiGMSSubclassProc, 1, 1);
                
                return 0.0;
                }`

This is the part that I use to load texture from the host program .. I just run it once in the host program.
        
        `ID3D11ShaderResourceView* my_texture = nullptr;
        int my_image_width = 0;
        int my_image_height = 0;
        fn_export double imgui_showimage() {

	

	bool ret = LoadTextureFromFile("e:\\mario1_input.png", &my_texture, &my_image_width, &my_image_height);
	IM_ASSERT(ret);
	
	return 1;
	
        }`

This is the part I use to show texture in the loop;

       `fn_export double imgui_imager() {
	ImGui::Text("pointer = %p", my_texture);
	ImGui::Text("size = %d x %d", my_image_width, my_image_height);
	ImGui::Image((void*)my_texture, ImVec2(my_image_width, my_image_height));
	
	return 1;
        }`

LoadTextureFromFile part is the same with example..

In addition, I added the result I got as a video ..

Do you have an opinion on this matter?

https://user-images.githubusercontent.com/76127603/115001915-c9ac2e00-9eac-11eb-95d4-56d0528abfd4.mp4

**Comment by ocornut** [comment]
Consider using RenderDoc (https://renderdoc.org/) to debug graphics issues such as DX11 ones.

**Comment by mbeytekin** [comment]
thank you so much. I hope it solves my problem :)

**Comment by mbeytekin** [comment]
hmm .. As I attached it to the host program, it only shows the operations on it. Anyway, I guess this topic is a little bit beyond me Thank you anyway for your attention.
